enginetest/queries/foreign_key_queries.go

```diff
@@ -1334,4 +1334,154 @@ var ForeignKeyTests = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "Table with inverted primary key referencing another table can insert rows",
+		SetUpScript: []string{
+			"create table a (x int, y int, primary key (x,y), INDEX `a_y_idx` (y));",
+			"create table b (x int, y int, primary key (y,x), foreign key (y) references a(y) on update cascade on delete cascade);",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:    "INSERT into a VALUES (1, 3);",
+				Expected: []sql.Row{{sql.NewOkResult(1)}},
+			},
+			{
+				Query:    "INSERT into b VALUES (2, 3);",
```

Comment:
can you include the field list for these insert statements pls. I'm assuming it's `(x,y)` for both table, but it would help to clarify